mage_ai/cache/tag.py

```diff
@@ -23,6 +27,34 @@ async def initialize_cache(self, replace: bool = False) -> 'TagCache':
     def get_tags(self) -> Dict:
         return self.get(self.cache_key) or {}
 
+    def get_tags_by_pipeline_uuid(self) -> Dict:
```

Comment:
We needed a cache for the `pipeline_uuid_to_tags_mapping` so that we can quickly filter pipeline runs by pipeline tags and also to display the pipeline tags for each pipeline run in the Pipeline Runs table. Not sure if it'd be better to create an entirely different cache class, but I added this method here since it was related to the `TagCache`.

Reply:
This method is actually not needed if you implement the pipeline run filtering by tags with my recommended way.

Reply:
@wangxiaoyou1993 I added a property `pipeline_tags` that uses this method to the PipelineRun [model](https://github.com/mage-ai/mage-ai/pull/3659/files#diff-81ce2b1693f591260806fa018d2c1f03076f636fafbc682bc6318d6d64d002cdR622) so that we can display the pipeline tags in the PipelineRuns table. I considered not displaying it in the table, but it seemed necessary to provide some assurance that the pipeline tag filter was properly filtering the pipeline runs and so that users can see which runs had certain pipeline tags if multiple tags were being filtered.
 
![image](https://github.com/mage-ai/mage-ai/assets/78053898/9507e7f4-3408-4efe-bc72-2006b20f2cef)

Reply:
You can just get the tags by pipeline using `pipeline.tags`

Reply:
Gotcha, i was overthinking it.